### PR TITLE
Finished signup route for backend

### PR DESCRIPTION
**Pull request by RafGDev**
I have finished the signup route. To test the app, use something like postman as the signup route is a post request.

I will probably modularise the code into smaller functions but for learning purposes, it is probably a better idea to have everything in the route so @jp-webdev can understand what is happening.

Maybe we could also add some automated tests later on?

### CODE REVIEW

File: backend/routes.js [hunk 1]
```diff
@@ -23,19 +34,75 @@ routes.getAllMessages = (req, res) => { // TYPE: GET
...
+  //If the username or pass is over 30 characters long
+  if (username.length > 30 || password.length > 30) {
+    res.json({"response": "error", "errorType": "tooLong"});
+    return;
+  }
+
+  //Creates a rando salt. Length of salt is 32 chars
+  const salt = crypto.randomBytes(16).toString("hex");
+  //Creates a sha1 hash of salt + password e.g. sha1(salt + password)
+  const hash = crypto.createHash('sha1').update(salt + password).digest('hex');
```
Comment (JGefroh) [inline backend/routes.js:70]: This can be changed later when it matters, but as a general reminder SHA1 is not a secure method to store passwords.

**Comment by jp-webdev** [comment]
For username and password -- I would recommended using some regex search patterns to ensure that the username and password follow a set of standards (a capital letter, a number, etc), (since it looks like all of the validation is happening on our side).

**Comment by IAmJohnOconnell** [comment]
Nice work so far. How do we go about incorporating this into the Front End?

**Comment by RafGDev** [comment]
@IAmJohnOconnell If you have a look on the other team's backend: https://github.com/space-turtles/issue-tracker/blob/issues-view/backend/server.js if you scroll to the bottom you can see that routes with /api prefixed are api routes. Then for any other routes, we can pass down the angular index.html file.

**Comment by RafGDev** [comment]
@jp-webdev could you implement that? I don't know regex at all. I could read up on it if you want me to.

**Pull request closed without merging**